MAX17055.device.lib.nut

```diff
@@ -0,0 +1,419 @@
+const MAX17055_DESIGN_CAP_REG         = 0x18;
+const MAX17055_CONFIG_REG             = 0x1D;
+const MAX17055_I_CHR_TERM_REG         = 0x1E;
+const MAX17055_AV_CAP_REG             = 0x1F;
+const MAX17055_TTF_REG                = 0x20;
+const MAX17055_DEV_NAME_REG           = 0x21;
+const MAX17055_V_EMPTY_REG            = 0x3A;
+const MAX17055_F_STAT_REG             = 0x3D;
+const MAX17055_DQ_ACC_REG             = 0x45;
+const MAX17055_DP_ACC_REG             = 0x46;
+const MAX17055_SOFT_WAKE_CMD_REG      = 0x60;
+const MAX17055_I_ALRT_TH_REG          = 0xB4;
+const MAX17055_HIB_CGF_REG            = 0xBA;
+const MAX17055_CONFIG_2_REG           = 0xBB;
+const MAX17055_MODEL_CFG_REG          = 0xDB;
+
+const MAX17055_DEFAULT_I2C_ADDR       = 0x6C;
+
+const MAX17055_SOFT_WAKE_CMD_CLEAR    = 0x0000;
+const MAX17055_SOFT_WAKE_CMD_WAKE     = 0x0090;
+const MAX17055_HIBERNATE_CMD_CLEAR    = 0x0000;
+
+const MAX17055_REG_CHECK_TIMEOUT_SEC  = 0.1;
+const MAX17055_REG_CHECK_NUM_RETRYS   = 20;
+const MAX17055_REG_VERIFY_TIMEOUT_SEC = 0.001;
+const MAX17055_REG_VERIFY_NUM_RETRYS  = 3;
+
+const MAX17055_CHRG_VOLT_DEFAULT_THRESH = 0xFF00;
+const MAX17055_TEMP_CURR_DEFAULT_THRESH = 0x7F80;
+
+const MAX17055_V_CHRG_4_2             = 0x00;
+const MAX17055_V_CHRG_4_4_OR_4_35     = 0x01;
+
+enum MAX17055_BATT_TYPE {
+    LiCoO2  = 0,
+    NCA_NCR = 2,
+    LiFePO4 = 6
+}
+
+class MAX17055 {
+
+    static VERSION = "1.0.0";
+
+    _i2c  = null;
+    _addr = null;
+
+    _capacityLSB = null;
+    _currLSB = null;
+
+    _regReadyCounter = 0;
+    _writeVerifyCounter = 0;
+
+    constructor(i2c, addr = null) {
+        _i2c = i2c;
+        _addr = (addr == null) ? MAX17055_DEFAULT_I2C_ADDR : addr;
+    }
+
+    // Currently only supports (EZ config not INI file, all calculations taken from software implementaion guide)
+    // Params:
+        // settings - table with required keys: desCap, senseRes, chrgTerm,
+            // emptyVTarget, recoveryV, chrgV, battType
+        // cb - callback function called when initialization finished
+    function init(settings, cb = null) {
+        // Make sure we have all required settings
+        if (settings.len() < 7 || !("desCap" in settings) || !("senseRes" in settings) ||
+            !("chrgTerm" in settings) || !("emptyVTarget" in settings) || !("recoveryV" in settings) ||
+            !("chrgV" in settings) || !("battType" in settings)) {
+            _handleErr("Missing a required setting. Cannot initialize MAX17055", cb);
+            return;
+        }
+
+        // Set Register Standard Resolution vars (these are based on resistor value)
+        _setLSBScalr(settings.senseRes);
+
+        // Get POR bit in status register (bit 1)
+        local status = _readReg(MAX17055_STATUS_REG, false);
+        if (status == null) {
+            return _handleErr(format("Error reading reg: 0x%02X Err: %i", MAX17055_STATUS_REG, _i2c.readerror()), cb);
+        } else if ((status & 0x0002) == 2) {
```

Comment:
No need to change it. I would rather write `} else if (status & 0x0002) {`, it's shorter and easier to understand.